website/src/content/development/forms.mdx

```diff
@@ -0,0 +1,540 @@
+      const data = formValuesToDoc(formValues);
+      // This would trigger the request, for example a mutation.
+      const result = await updateChannel(data);
+      // If successful, show a notification.
+      // If errored, show an error notification.
+    },
+    [updateChannel]
+  );
+
+  if (!channel) {
+    return <LoadingSpinner />;
+  }
+
+  return (
+    <Spacings.Stack scale="xl">
+      <Text.Headline as="h1">
+        Manage Channel
+      </Text.Headline>
+      <ChannelsForm
+        initialValues={docToFormValues(channel)}
+        onSubmit={handleSubmit}
+      />
+    </Spacings.Stack>
+  );
+}
+```
+
+## Using modal pages
+
+Most of the time a create or details page with a form can be implemented using either the [FormModalPage](/api-reference/commercetools-frontend-application-components#formmodalpage) or the [CustomFormModalPage](/api-reference/commercetools-frontend-application-components#customformmodalpage) components.
+
+Using these components has the advantage of providing the form control buttons (for example `Cancel` and `Save`) in the correct place, according to our design guidelines.
+
+However, the form component must be defined "outside" of the modal page to be able to pass the necessary functions to the modal page to interact with the form.
+
+Therefore, our Channels form must be refactored to defined all the form elements but return them using the function-as-child component pattern.
+
+```jsx title="channels-form.js"
+import { useFormik } from 'formik';
+import LocalizedTextField from '@commercetools-uikit/localized-text-field';
+import TextField from '@commercetools-uikit/text-field';
+import Spacings from '@commercetools-uikit/spacings';
+import validate from './validate';
+
+const ChannelsForm = (props) => {
+  const formik = useFormik({
+    // Pass initial values from the parent component.
+    initialValues: props.initialValues,
+    // Handle form submission in the parent component.
+    onSubmit: props.onSubmit
+    validate,
+    enableReinitialize: true,
+  });
+
+  // Only contains the form elements, no buttons.
+  const formElements = (
+    <Spacings.Stack scale="l">
+      <LocalizedTextField
+        name="name"
+        title="Name"
+        isRequired={true}
+        selectedLanguage="en"
+        value={formik.values.name}
+        errors={formik.errors.name}
+        touched={formik.touched.name}
+        onChange={formik.handleChange}
+        onBlur={formik.handleBlur}
+      />
+      <TextField
+        name="key"
+        title="Key"
+        isRequired={true}
+        value={formik.values.key}
+        errors={formik.errors.key}
+        touched={formik.touched.key}
+        onChange={formik.handleChange}
+        onBlur={formik.handleBlur}
+      />
+    </Spacings.Stack>
+  );
+
+  return props.children({
+    formElements,
+    isDirty: formik.dirty,
+    isSubmitting: formik.isSubmitting,
+    submitForm: formik.handleSubmit,
+    handleCancel: formik.handleReset,
+  });
```

Comment:
Couldn't this be a hook? `useChannelsForm`? 🤔

Reply:
Hmm could be, although it's weird to return React elements from a hook.

Also, you can't always use `useFormik`, as using field arrays requires to use the `<Formik>` component.

Reply:
Yip. Was just thinking out loud case FaaC feels odd these days at times.